### PR TITLE
Leader mapper new functions

### PR DESCRIPTION
**Pull request by Lehersun**
- LeaderMapper:update - update Leader data in a database;
- LeaderMapper:delete - delete the data about Leader from a database by id

### CODE REVIEW

File: tests/integration/DataMapper/LeaderMapperTest.php [hunk 1]
```diff
@@ -38,4 +38,38 @@ public function testFind()
...
+
+    /**
+     * @throws \Doctrine\DBAL\Exception
+     */
+    public function testUpdate()
+    {
+        $mapper = new LeaderMapper(self::$db);
+        $newLeader = $mapper->find(1);
+        $newLeader->setName('Vasiliy');
+        $newLeader->setSurname('Chapaev');
```
Comment (Magomogo) [inline tests/integration/DataMapper/LeaderMapperTest.php]: Not clear here, do you read someone from the DB and the make it be some other person? Setters are evil: they mutate an object making things more complex than they are.

In reality you want to produce a new object and use it to update the database:

```
$mapper->update(new Leader({data from the network}), $id);
```

**Review by Magomogo — CHANGES_REQUESTED** [review]
Please implement without the setters.

File: tests/integration/DataMapper/LeaderMapperTest.php (hunk 1 above)
Comment (Lehersun) [reply tests/integration/DataMapper/LeaderMapperTest.php]: Setters are evil - good theme for a poster near a work place

Comment (Magomogo) [reply tests/integration/DataMapper/LeaderMapperTest.php]: :)

**Pull request merged**